contacts/contact_custom_report.sql

```diff
@@ -0,0 +1,549 @@
+
+  legal_guardian_2.physical_address_line_1 AS "Legal Guardian 2 Address Line 1",
+  legal_guardian_2.physical_address_line_2 AS "Legal Guardian 2 Address Line 2",
+  legal_guardian_2.physical_address_city AS "Legal Guardian 2 City",
+  legal_guardian_2.physical_address_state AS "Legal Guardian 2 State",
+  legal_guardian_2.physical_address_zip AS "Legal Guardian 2 Zip",
+
+  legal_guardian_2.mailing_address_line_1 AS "Legal Guardian 2 Mailing Address Line 1",
+  legal_guardian_2.mailing_address_line_2 AS "Legal Guardian 2 Mailing Address Line 2",
+  legal_guardian_2.mailing_address_city AS "Legal Guardian 2 Mailing City",
+  legal_guardian_2.mailing_address_state AS "Legal Guardian 2 Mailing State",
+  legal_guardian_2.mailing_address_zip AS "Legal Guardian 2 Mailing Zip",
+
+-- Emergency Contact 1
+  emergency_contact_1.last_name AS "Emergency Contact 1 Last Name",
+  emergency_contact_1.first_name AS "Emergency Contact 1 First Name",
+  emergency_contact_1.contact_type AS "Emergency Contact 1 Contact Type",
+
+  emergency_contact_1.email_address AS "Emergency Contact 1 Email",
+
+  emergency_contact_1.cell_phone_number AS "Emergency Contact 1 Cell Phone",
+  emergency_contact_1.home_phone_number AS "Emergency Contact 1 Home Phone",
+  emergency_contact_1.work_phone_number AS "Emergency Contact 1 Work Phone",
+
+-- Emergency Contact 1
+  emergency_contact_2.last_name AS "Emergency Contact 2 Last Name",
+  emergency_contact_2.first_name AS "Emergency Contact 2 First Name",
+  emergency_contact_2.contact_type AS "Emergency Contact 2 Contact Type",
+
+  emergency_contact_2.email_address AS "Emergency Contact 2 Email",
+
+  emergency_contact_2.cell_phone_number AS "Emergency Contact 2 Cell Phone",
+  emergency_contact_2.home_phone_number AS "Emergency Contact 2 Home Phone",
+  emergency_contact_2.work_phone_number AS "Emergency Contact 2 Work Phone"
+
+FROM student_session_aff AS enrollments
+  LEFT JOIN sessions ON enrollments.session_id = sessions.session_id
+
+  -- Get student info
+  LEFT JOIN public.students stud USING(student_id)
+  LEFT JOIN public.sites AS st ON sessions.site_id = st.site_id
+
+  -- Get mentor
+  LEFT JOIN student_house_aff ON stud.student_id = student_house_aff.student_id
+    AND student_house_aff.session_id = sessions.session_id
+  LEFT JOIN houses ON houses.house_id = student_house_aff.house_id
+
+  -- Get contact info
+  LEFT JOIN legal_guardian_1 ON legal_guardian_1.student_id = enrollments.student_id
+  LEFT JOIN legal_guardian_2 ON legal_guardian_2.student_id = enrollments.student_id
+  LEFT JOIN emergency_contact_1 ON emergency_contact_1.student_id = enrollments.student_id
+  LEFT JOIN emergency_contact_2 ON emergency_contact_2.student_id = enrollments.student_id
+
+WHERE
+  -- get students enrolled in the current academic year
+  sessions.academic_year = 2018
+
+  -- get students enrolled in a window around the current time
+  AND enrollments.entry_date <= current_date
```

Comment:
I'm not sure how filtering the set of enrolled students here is going to work when we send it to Illuminate.

I'm curious if there was a specific reason why you chose to pull students from student_session_aff and use the enrollment day over using the ss_current. For most use cases I definitely prefer this way of doing things, but perhaps in this use case it makes less sense.

Reply:
I don't think that you need to swap it out for a variable, I think that their support desk does so.

Reply:
Filtering -  I think it makes sense to ask Illuminate to have this report be reactive to the control panel (if you are in Tahoma on 4/16, you only see Tahoma students enrolled on 4/16).  What do you think?

Student_session_aff -  I did not edit the student_session_aff join.  You had that in there already.  

Data quality - I checked for duplicates in Excel.  Then I compared count to Search 2.0.  I just did this again to confirm.  CA Search 2.0 differs by 2 because I am filtering out Howardo Shenny at SPS Tour and Nolan Bell at Summit NPS.  I checked with Dan and Maddy on that and they said he should not appear on this report since he is not onsite and this is most often used for onsite communications.  Regardless, if Illuminate implements control panel filtering, then he will show on the report when people are in Summit NPS.

Reply:
Filtering - I don't think that is how this works, but you can ask.

Student_session_aff - It may make sense to change this to ss_current so that you don't have to select the contacts as of data when running the report. What do you think?

Data quality - If you've checked and feel confident, then I'm ok with this.